src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -193,7 +193,30 @@ private void RecordReadInLocalFunction(int slot)
             Debug.Assert(localFunc != null);
 
             var usages = GetOrCreateLocalFuncUsages(localFunc);
-            usages.ReadVars[slot] = true;
+
+            // If this slot is a struct with individually assignable
+            // fields we need to record each field assignment separately,
+            // since some fields may be assigned when this read is replayed
+            VariableIdentifier id = variableBySlot[slot];
+            var type = VariableType(id.Symbol);
```

Comment:
No. `VariableIdentifier` has Asserts in the constructor that validate that `symbol` is not null.